src/Uno.UI/UI/Xaml/GestureHandler.Android.cs

```diff
@@ -12,135 +12,44 @@ namespace Windows.UI.Xaml
-		}
-		#endregion
-
-		#region DoubleTapped event
-		private event DoubleTappedEventHandler DoubleTapped;
-
-		internal void RegisterDoubleTapped(DoubleTappedEventHandler handler)
-		{
-			DoubleTapped += handler;
-			SetShouldHandleDoubleTap(true);
-		}
-
-		internal void UnregisterDoubleTapped(DoubleTappedEventHandler handler)
-		{
-			DoubleTapped -= handler;
-			SetShouldHandleDoubleTap(DoubleTapped != null);
-		}
-		#endregion
-
-		#region PointerPressed event
-		private event PointerEventHandler PointerPressed;
-
-		internal void RegisterPointerPressed(PointerEventHandler handler)
-		{
-			PointerPressed += handler;
-			SetShouldHandleDown(true);
-		}
-
-		internal void UnregisterPointerPressed(PointerEventHandler handler)
+		internal void UpdateShouldHandle(RoutedEvent routedEvent, bool shouldHandle)
 		{
-			PointerPressed -= handler;
-			SetShouldHandleDown(PointerPressed != null);
-		}
-		#endregion
-
-		#region PointerReleased event
-		private event PointerEventHandler PointerReleased;
-
-		internal void RegisterPointerReleased(PointerEventHandler handler)
-		{
-			PointerReleased += handler;
-			SetShouldHandleUp(true);
-		}
-
-		internal void UnregisterPointerReleased(PointerEventHandler handler)
-		{
-			PointerReleased -= handler;
-			SetShouldHandleUp(PointerReleased != null);
-		}
-		#endregion
-
-		#region PointerMoved event
-		private event PointerEventHandler PointerMoved;
-
-		internal void RegisterPointerMoved(PointerEventHandler handler)
-		{
-			PointerMoved += handler;
-			SetShouldHandleMove(true);
-		}
-
-		internal void UnregisterPointerMoved(PointerEventHandler handler)
-		{
-			PointerMoved -= handler;
-			SetShouldHandleMove(PointerMoved != null);
-		}
-		#endregion
-
-		#region PointerExited event
-		private event PointerEventHandler PointerExited;
-
-		internal void RegisterPointerExited(PointerEventHandler handler)
-		{
-			PointerExited += handler;
-			SetShouldHandleExit(true);
-		}
-
-		internal void UnregisterPointerExited(PointerEventHandler handler)
-		{
-			PointerExited -= handler;
-			SetShouldHandleExit(PointerExited != null);
-		}
-		#endregion
-
-		#region PointerEntered event
-		private event PointerEventHandler PointerEntered;
-
-		internal void RegisterPointerEntered(PointerEventHandler handler)
-		{
-			PointerEntered += handler;
-			SetShouldHandleEnter(true);
-		}
-
-		internal void UnregisterPointerEntered(PointerEventHandler handler)
-		{
-			PointerEntered -= handler;
-			SetShouldHandleEnter(PointerEntered != null);
-		}
-		#endregion
-
-		#region PointerCanceled event
-		private event PointerEventHandler PointerCanceled;
-
-		internal void RegisterPointerCanceled(PointerEventHandler handler)
-		{
-			PointerCanceled += handler;
-			SetShouldHandleCancel(true);
-		}
-
-		internal void UnregisterPointerCanceled(PointerEventHandler handler)
-		{
-			PointerCanceled -= handler;
-			SetShouldHandleCancel(PointerCanceled != null);
+			// TODO: How expensive is this?
```

Comment:
Is the comment still valid? (If it's referring to the 'if else' chain, I think we can live with it)

Reply:
It's reffering of one point moved from the PR to issue #322 :
> _Implement a WPF-inspired internal virtual RoutedEventArgs.InvokeEventHandler(Delegate handler, object target) to avoid switch cases in UIElement.InvokeHandler_